### PR TITLE
Add battery voltage logic to fancy-node

### PR DESCRIPTION
**Pull request by ademuri**
This does not yet display the battery status at all - how do you think it's best to do that?

### CODE REVIEW

**Comment by wmmiii** [comment]
> This does not yet display the battery status at all - how do you think it's best to do that?

I was thinking of displaying it on the strips at startup or using the on-board LED to periodically blink red when you're running low.

I'm on mobile so I won't review right now.

**Comment by ademuri** [comment]
Updated to display the battery status on the onboard LED for the first 20 seconds.

File: src/devices/fancy-node/fancy-node.cc [hunk 1]
```diff
@@ -46,6 +69,28 @@ void setup() {
...
+    led_manager->SetOnboardLed(CRGB(4, 0, 0));
+  } else {
+    // Display battery status on onboard LED for the first 20 seconds
+    if (millis() < 20 * 1000) {
+      uint16_t battery_val = battery_average_filter.GetFilteredValue() -
+                             BatteryVoltageToRawReading(kBatteryDead);
+      uint16_t battery_range = BatteryVoltageToRawReading(kBatteryFull) -
+                               BatteryVoltageToRawReading(kBatteryDead);
+      // Red is hue 0
+      uint8_t hue = (battery_val * HUE_GREEN) / battery_range;
```
Comment (wmmiii) [inline src/devices/fancy-node/fancy-node.cc:90]: Nice animation! I think we should shove this in the LedManager startup animation however lets not worry about that right now. (I don't know when we'd do voltage sampling in that case.

File: src/devices/fancy-node/fancy-node.cc [hunk 2]
```diff
@@ -12,26 +15,46 @@
...
+ExponentialMovingAverageFilter<uint16_t> battery_average_filter{
+    []() { return battery_median_filter.GetFilteredValue(); },
+    kBatteryFilterAlpha};
+
+static constexpr float kBatteryDead = 3.72;
+static constexpr float kBatteryFull = 4.2;
+
+// voltage = battery * 180 / (62 + 180) * 3.3 / 1024
+constexpr uint16_t BatteryVoltageToRawReading(float voltage) {
+  return voltage / (180 * 3.3 / (62 + 180) / 1024);
```
Comment (wmmiii) [inline src/devices/fancy-node/fancy-node.cc]: These are some magic numbers, move them into constexpr variables instead?

**Review by wmmiii — APPROVED** [review]
LGTM with some nits

File: src/devices/fancy-node/fancy-node.cc (hunk 2 above)
Comment (ademuri) [reply src/devices/fancy-node/fancy-node.cc]: Done.

File: src/devices/fancy-node/fancy-node.cc (hunk 1 above)
Comment (ademuri) [reply src/devices/fancy-node/fancy-node.cc:90]: Ack

**Pull request merged**